content/crutchcorn/collections/art-of-accessibility/posts/art-of-a11y-text/index.md

```diff
@@ -0,0 +1,561 @@
+
+```javascript
+const CONTRAST_RATIO = (L1 + 0.05) / (L2 + 0.05)
+```
+
+> L1 is the relative luminance of the lighter of the colors.
+> L2 is the relative luminance of the darker of the colors.
+
+Or manually, using various tools such as [WebAIM's Color Contrast Checker](https://webaim.org/resources/contrastchecker/):
+
+![A contrast checker UI showing the contrast between a blue and a black color](./webaim_checker.png)
+
+## CSS Detection
+
+When comparing AA to AAA compliance, a common qualm lodged against AAA is that its contrast requirements feel too constrictive for a site's "look and feel" compared to AA's easier compliance ratios.
+
+Luckily, there are ways we can have both!
+
+Users are able to opt into an operating system's "high contrast" mode, which will notify the browser to follow in suit.
+
+![Windows 11's "Contrast themes" showing the different styling of high-contrast modes](./windows_11_high_contrast.png)
+
+![macOS configured to show "increased contrast" with more defined borders and outlines](./mac_os_high_contrast.png)
+
+This, in turn, is exposed to our site's CSS through [the `prefers-contrast` media query](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-contrast) :
+
+```css
+@media (prefers-contrast: more) {
+  .contrast {
+    color: black;
+    background: #f0efef;
+  }
+}
+```
+
+-----
+
+We can even use JavaScript to detect this media query from inside of our runtime code:
+
+```javascript
+// Use JS to detect the user's preference for contrast
+const mediaQuery = window.matchMedia("(prefers-contrast: more)");
+
+function changeText(matches) {
+	if (matches) {
+		el.textContent = "The user prefers more contrast";
+	} else {
+		el.textContent = "The user has not specified a preference for contrast";
+	}
+}
+
+// To check the initial value:
+changeText(mediaQuery.matches);
+
+// To listen for changes:
+mediaQuery.addEventListener("change", (e) => {
+	changeText(e.matches);
+});
+```
+
+<iframe data-frame-title="HTML CSS Detection - StackBlitz" src="pfp-code:./art-of-a11y-html-contrast-detection-2?embed=1&file=src/main.js" sandbox="allow-modals allow-forms allow-popups allow-scripts allow-same-origin"></iframe>
+
+# Text Size
+
+Let's do a similar demo to the one above. Can you read this?
+
+<p style="font-size: 0.5rem">This is very small text</p>
+
+> There is text above this that says "This is very small text".
+
+How about this?
+
+<p style="font-size: 0.5rem; filter: blur(2px)">This is very small text</p>
+
+> The same text is now blurred.
+
+Probably not? Alright! Now how about _this_ one?
+
+<p style="font-size: 3rem; filter: blur(2px)">This is large text</p>
+
+> It now says "This is large text"!
+
+This demonstrates how important text sizing is for so many of our users. 
+
+## `rem` vs `px` Values
+
+In web development, we often use `px` to represent sizing of things. `1px` _roughly_ means `1 pixel`, which works fine for many things on the web.
```

Comment:
Only if we had a deep dive article into device:pixel ratio, viewport sizing, and zooming lol